server.Dockerfile

```diff
@@ -14,21 +14,19 @@ RUN (cd ./temporal && go mod download all)
 COPY ./tctl/go.mod ./tctl/go.sum ./tctl/
 RUN (cd ./tctl && go mod download all)
 
-# cache Temporal CLI packages as a docker layer
-COPY ./cli/go.mod ./cli/go.sum ./cli/
-RUN (cd ./cli && go mod download all)
+# cache Temporal CLI binary
+RUN set -e; curl -sSf https://temporal.download/cli.sh | sh /dev/stdin --dir ./cli
```

Comment:
btw, I'm not a fan of the `curl ... | sh` approach at all. That is one big security hole.

Reply:
resolved under a different comment

Reply:
sh -c is the same thing. I wouldn't say it's a "hole" since we control everything. But it could be cleaner to inline the install commands here instead of using the script. It should just be a few commands. Then you can just do the version thing here and don't have to wait for the other PR